### PR TITLE
Remove AA >: A constraint from XorT.orElse

### PR DESCRIPTION
**Pull request by ceedubs**

### CODE REVIEW

**Comment by adelbertc** [comment]
:+1:

File: core/src/main/scala/cats/data/XorT.scala [hunk 1]
```diff
@@ -29,11 +29,11 @@ final case class XorT[F[_], A, B](value: F[A Xor B]) {
     }
   }
 
-  def orElse[AA >: A, BB >: B](default: => XorT[F, AA, BB])(implicit F: Monad[F]): XorT[F, AA, BB] = {
+  def orElse[AA, BB >: B](default: => XorT[F, AA, BB])(implicit F: Monad[F]): XorT[F, AA, BB] = {
     XorT(F.flatMap(value) { xor =>
       xor match {
         case Xor.Left(_) => default.value
```
Comment (fthomas) [inline core/src/main/scala/cats/data/XorT.scala]: Since we are not using `xor` after `match` I think we can just remove `xor => xor match {` ... `}`.

Comment (ceedubs) [inline core/src/main/scala/cats/data/XorT.scala]: Yet another good catch. I pushed an update.

**Comment by fthomas** [comment]
:+1:

**Pull request merged**